src/common/snippets/tests/src/utils.cpp

```diff
@@ -22,7 +25,36 @@ TokenizationConfig get_default_tokenization_config() {
 }
 
 CommonOptimizations::Config get_default_common_optimizations_config() {
-    static const CommonOptimizations::Config conf(1, true);
+    static CommonOptimizations::Config conf(1, true);
+    static bool initialized = false;
+    if (!initialized) {
+        conf.set_transpose_support_callback([](const std::shared_ptr<const ov::Node>& node) -> bool {
```

Comment:
Ideally, we need to share the callback code between tests and plugin transformation pipeline. Maybe we can introduce some helper for that? The helper could have e.g. bool parameter which defines if the "is_brgemm_case" check should be applied as well

What do you think?